pycheribuild/projects/project.py

```diff
@@ -1293,6 +1295,27 @@ def update(self, current_project: "Project", *, src_dir: Path, default_src_dir:
                     if current_project.query_yes_no("Update to correct URL?"):
                         runCmd("git", "remote", "set-url", "origin", self.url, runInPretendMode=True, cwd=src_dir)
 
+        # see if we have an upstream
+        head = runCmd("git", "symbolic-ref", "-q", "HEAD",
+                      captureOutput=True, cwd=src_dir, print_verbose_only=True, runInPretendMode=True).stdout.strip().decode("utf-8")
+        upstream_branch = runCmd("git", "for-each-ref", '--format=%(upstream:short)', head, "origin/mainline",
+                      captureOutput=True, cwd=src_dir, print_verbose_only=True, runInPretendMode=True).stdout.strip().decode("utf-8")
```

Comment:
A bit more digging suggested this might be a better command `git rev-parse --abbrev-ref --symbolic-full-name @{upstream}`.  I'm not sure when either set of commands became available.

Reply:
Right now I think the oldest OS that we try to support is Ubuntu 16.04. I just managed to run both of these on a Ubuntu 16.04 docker container, so I think the `git rev-parse --abbrev-ref --symbolic-full-name @{upstream}` command looks good.

Reply:
bah, upon further checking, that command fails with an error if no upstream is present and thus doesn't work.  I've reverted the previous change so we're back to the original.

Reply:
Maybe the following works? `rev-parse` returns 128 for me if there is no upstream, but I'm not sure if that is always the case, so I added the output check too:
```python
try:
  upstream_branch = runCmd("git", "rev-parse", "--abbrev-ref", "--symbolic-full-name", "@{upstream}",
       captureError=True, captureOutput=True, cwd=src_dir, print_verbose_only=True,
        runInPretendMode=True).stdout.strip().decode("utf-8")
except subprocess.CalledProcessError as e:
  if e.returncode == 128 or "no upstream configured" in e.stderr:
    print(coloured(AnsiColour.blue, "No upstream to update from"))
    return
  raise
```